### PR TITLE
feat: report skybox time to kernel

### PR DESCRIPTION
**Pull request by pbosio**

### CODE REVIEW

File: unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    public void Configure(float normalizationFactor, float cycle)
+    {
+        timeNormalizationFactor = normalizationFactor;
+        cycleTime = cycle;
+    }
+
+    public void ReportTime(float time)
+    {
+        bool isPaused = !DataStore.i.skyboxConfig.useDynamicSkybox.Get();
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs:31]: I find this inverse logic hard to follow. I would replace `isPaused` and `wasPaused` for `active` `wasActive` and if changing kernel is bothersome, do the negation in `WebInterface.ReportTime`.

File: unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        // NOTE: if not paused and pause state didn't change there is no need to report
+        // current time since it's being calculated on kernel side
+        if (!isPaused && !wasPaused)
+        {
+            return;
+        }
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs:38]: Cannot we just report when the time has manually changed or the skybox has been paused/resumed?

With a timestamp and an offset we should be able to calculate the rest.

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs [hunk 3]
```diff
@@ -44,6 +44,9 @@ public class SkyboxController : IPlugin
         private int syncCounter = 0;
         private int syncAfterCount = 10;
 
+        // Report to kernel
+        private TimeReporter timeReporter = new TimeReporter();
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs:48]: I would extract an interface for `TimeReporter` so we can mock it.

File: unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs (hunk 2 above)
Comment (AjimenezDCL) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs:38]: Okay, I think we can forget the above, unless I'm mistaken, it's already coded that way.

**Review by AjimenezDCL — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs (hunk 2 above)
Comment (pbosio) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs]: yeap, it's already coded that way :)

File: unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs (hunk 1 above)
Comment (pbosio) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/TimeReporter/TimeReporter.cs]: done

File: unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs (hunk 3 above)
Comment (pbosio) [reply unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs]: done

**Review by pravusjif — APPROVED** [review]
LGTM

**Review by AjimenezDCL — APPROVED** [review]
